bundle/permissions/utils.go

```diff
@@ -0,0 +1,81 @@
+package permissions
+
+import (
+	"context"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/libs/diag"
+)
+
+func convert(
+	ctx context.Context,
+	bundlePermissions []resources.Permission,
+	resourcePermissions []resources.Permission,
+	resourceName string,
+	lm map[string]string,
+) []resources.Permission {
+	permissions := make([]resources.Permission, 0)
+	for _, p := range bundlePermissions {
+		level, ok := lm[p.Level]
+		// If there is no bundle permission level defined in the map, it means
+		// it's not applicable for the resource, therefore skipping
+		if !ok {
+			continue
+		}
+
+		if notifyForPermissionOverlap(ctx, p, resourcePermissions, resourceName) {
+			continue
```

Comment:
We should consider overriding permissions here if bundle permissions are larger in scope than the resource permissions. 

Consider the case:
```
permissions:
  user_name: abc
  level: CAN_MANAGE

jobs:
  foo:
     permissions: {
        user_name: "abc"
        level: CAN_VIEW
    },
```

In this case the correct behavior IMO might be for `abc` to have CAN_MANAGE access to the job. I guess the intention here is to provide users the ability to downgrade permission scope per resource?

Reply:
@shreyas-goenka not really, as per discussuions with Pieter, Lennart and Fabian we agreed that at this momemnt if such overlap occurs we just want to notify customers about to avoid any unambiguity in permissions and potential errors in configurations.